gradle/scripts/lib/java.gradle

```diff
@@ -124,12 +119,16 @@ configure(projectsWithFlags('java')) {
 
     // Set the sensible compiler options.
     tasks.withType(JavaCompile) {
-        def targetJavaVersionStr = JavaVersion.toVersion(targetJavaVersion).toString()
-        sourceCompatibility = project.findProperty('javaSourceCompatibility') ?: targetJavaVersionStr
-        targetCompatibility = project.findProperty('javaTargetCompatibility') ?: targetJavaVersionStr
+        def targetJavaVersionStr = null
+        if (targetJavaVersion != null) {
+            targetJavaVersionStr = JavaVersion.toVersion(targetJavaVersion).toString()
+        }
+        sourceCompatibility = targetJavaVersionStr ?: project.findProperty('javaSourceCompatibility') ?: '1.8'
```

Comment:
`javaN` flag and `javaSourceCompatibility` are mutually exclusive. I thought we may give `javaN` higher priority over `javaSourceCompatibility`. Because `javaSourceCompatibility` can be configured globally for all modules but `javaN` flag can be set to each module level.

Reply:
That makes sense. Thanks for the explanation. 🙇